packages/scene-composer/src/utils/objectThreeUtils.ts

```diff
@@ -68,8 +68,34 @@ export function enableShadow(component: IModelRefComponentInternal, obj: THREE.O
   }
 }
 
-export const resetObjectCenter = (object: THREE.Object3D) => {
-  const box = new THREE.Box3().setFromObject(object);
-  box.getCenter(object.position);
-  object.position.multiplyScalar(-1);
+export const resetObjectCenter = (obj: THREE.Object3D) => {
+  const box = new THREE.Box3().setFromObject(obj);
+  box.getCenter(obj.position);
+  obj.position.multiplyScalar(-1);
+};
+
+export const getSafeBoundingBox = (obj: THREE.Object3D): THREE.Box3 => {
+  // Because LineSegments have absurd sizes in ThreeJS we need to account for these in the scene and ignore them.
+  // Map all Line Segments to their parent for re-parenting
+  const lineMap: Map<THREE.Object3D, THREE.Object3D> = new Map<THREE.Object3D, THREE.Object3D>();
+  obj.traverse((child) => {
+    if ((child as THREE.LineSegments).isLineSegments) {
+      if (child.parent) {
```

Comment:
No need to nest. This can be expressed as a single `if` check. `if (child as THREE.LineSegments).isLineSegments && child.parent) ...`